src/Plugin/resource/Field/ResourceFieldCollection.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\restful\Plugin\resource\Field\ResourceFieldCollection.
+ */
+
+namespace Drupal\restful\Plugin\resource\Field;
+
+class ResourceFieldCollection implements ResourceFieldCollectionInterface {
```

Comment:
This one seems to be related to entity fields. The problem I see is for other providers (e.g. db), we will have to repeat some of the definitions